oap-server/analyzer/agent-analyzer/src/main/java/org/apache/skywalking/oap/server/analyzer/provider/trace/TraceSamplingPolicyWatcher.java

```diff
@@ -0,0 +1,217 @@
+    private final AnalyzerModuleConfig moduleConfig;
+
+    public TraceSamplingPolicyWatcher(AnalyzerModuleConfig moduleConfig, ModuleProvider provider) {
+        super(AnalyzerModule.NAME, provider, "traceSamplingPolicy");
+        this.moduleConfig = moduleConfig;
+        loadDefaultPolicy();
+    }
+
+    @Override
+    public void notify(ConfigChangeEvent value) {
+        if (EventType.DELETE.equals(value.getEventType())) {
+            this.settingsString.set("");
+            log.info("[trace-sampling-policy] Delete trace-sample-policy,use default config");
+            loadDefaultPolicy();
+        } else {
+            activeSetting(value.getNewValue());
+        }
+    }
+
+    @Override
+    public String value() {
+        return this.settingsString.get();
+    }
+
+    public SamplePolicy getSamplePolicy(String service) {
+        return this.samplePolicySettings.get().getPolicy(service);
+    }
+
+    public int getSampleRate() {
+        return this.samplePolicySettings.get().global.rate;
+    }
+
+    public int getSlowTraceSegmentThreshold() {
+        return this.samplePolicySettings.get().global.duration;
+    }
+
+    public boolean shouldSample(int duration) {
+        return (this.samplePolicySettings.get().global.duration > -1) && (duration >= this.samplePolicySettings.get().global.duration);
+    }
+
+    private void loadDefaultPolicy() {
+        SamplePolicySettings rateSetting = parseFromFile(moduleConfig.getTraceSamplingPolicySettingsFile());
+        // If settingFile has a empty config, use the default settings
+        this.samplePolicySettings.set(rateSetting == null ? defaultSampleConfigSettings() : rateSetting);
+        log.info("[trace-sampling-policy] Default configured trace-sample-policy: {}", this.samplePolicySettings);
+    }
+
+    private void activeSetting(String config) {
+        if (log.isDebugEnabled()) {
+            log.debug("[trace-sampling-policy] Updating using new config: {}", config);
+        }
+        onUpdated(parseFromYml(config));
+    }
+
+    private void onUpdated(final SamplePolicySettings samplePolicySettings) {
+        if (!isNull(samplePolicySettings)) {
+            this.samplePolicySettings.set(samplePolicySettings);
+            log.info("[trace-sampling-policy] Updating trace-sample-policy with: {}", samplePolicySettings);
+        } else if (StringUtil.isBlank(this.settingsString.get())) {
+            log.info("[trace-sampling-policy] Trace-sample-policy been set empty string,use default config");
+            loadDefaultPolicy();
+        } else {
+            log.info("[trace-sampling-policy] Parse yaml fail, retain last configuration: {}", this.samplePolicySettings);
+        }
+    }
+
+    private SamplePolicySettings parseFromFile(final String file) {
+        try {
+            final Reader reader = ResourceUtils.read(file);
+            Map<String, Object> map = new Yaml(new ClassFilterConstructor(new Class[]{
```

Comment:
This is a wrong usage of API. You should use `SafeConstructor`. Take `RulesReader` as a reference. 
The `ClassFilterConstructor` is designed for requiring initialization from customer Java class.